src/server/req_rescq.c

```diff
@@ -338,7 +338,7 @@ req_confirmresv(struct batch_request *preq)
 	 * is invalid, set it to some time after the soonest occurrence is to start
 	 */
 	if (strcmp(preq->rq_extend, PBS_RESV_CONFIRM_FAIL) == 0) {
-		if (is_degraded) {
+		if (is_degraded && !is_being_altered) {
```

Comment:
What happens if you alter a degraded reservation?

Reply:
This change is just for that case (altering a degraded reservation). It should be treated that the alter request has failed.

Reply:
OK I walked through the logic.  It looks good.